src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/finalparameters/Example4.java

```diff
@@ -0,0 +1,45 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="FinalParameters">
+      <property name="tokens" value="FOR_EACH_CLAUSE, LITERAL_CATCH"/>
+      <property name="ignoreUnnamedParameters" value="true"/>
+    </module>
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.finalparameters;
+
+// xdoc section -- start
+public class Example4 {
+
+  void testCatchParameters() {
+    try {
+      int x = 1 / 0;
+    }
+    catch (Exception e) { // violation, 'Parameter e should be final'
+      System.out.println(e);
+    }
+    try {
+      int x = 1 / 0;
+    }
+    catch (Exception _) { // ok,unnamed catch parameter is ignored, it can't be used
+      System.out.println("infinity");
+    }
+  }
+
+  void testForEachParameters() {
+    int[] l = {1, 2, 3};
+    int x = 0;
+    for (int number: l) { // violation, 'Parameter number should be final'
+      System.out.println(number);
+    }
+    for (int _: l) {  // ok, unnamed catch parameter is ignored, it can't be used
```

Comment:
Looks like copypasta
```suggestion
    for (int _: l) {  // ok, unnamed enhanced for loop variable, it is implicitly final
```

Reply:
> Looks like copypasta

Yes :) thank you. Updated the above comment also to be consistent in wording.
`it can't be used` =>` it is implicitly final`